rs/moq-native/src/server.rs

```diff
@@ -301,20 +338,20 @@ impl QuicRequest {
 
 #[derive(Debug)]
 struct ServeCerts {
-	certs: Vec<Arc<CertifiedKey>>,
+	certs: RwLock<Vec<Arc<CertifiedKey>>>,
 	provider: crypto::Provider,
 }
 
 impl ServeCerts {
 	pub fn new(provider: crypto::Provider) -> Self {
 		Self {
-			certs: Vec::new(),
+			certs: RwLock::new(Vec::new()),
 			provider,
 		}
 	}
 
-	// Load a certificate and corresponding key from a file
-	pub fn load(&mut self, chain: &PathBuf, key: &PathBuf) -> anyhow::Result<()> {
+	// Load a certificate and corresponding key from a file, but don't add it to the certs
```

Comment:
Why not? Why is this is `&mut self` if it doesn't even mutate `ServeCerts`?

Reply:
It was `&mut self` but I've removed that because on reload it first loads all the certificates/keys and then replaces them. This prevents that certificates get unloaded when there is an issue with the new certificates like a key that doesn't match the certificate.